static/js/filter.js

```diff
@@ -99,6 +99,17 @@ function message_matches_search_term(message, operator, operand) {
     case 'sender':
         return people.id_matches_email_operand(message.sender_id, operand);
 
+    case 'huddle-with':
+        var operand_id = people.get_user_id(operand);
+        if (!operand_id) {
+            return false;
+        }
+        var message_ids = people.huddle_with_user_ids(message);
+        if (!message_ids) {
+            return false;
+        }
+        return (operand_id in message_ids);
+
```

Comment:
Add a comment here explaining that in theory we should check if the current user was in the list, but that is implied by the fact that the current user has access to the message.